app/javascript/src/Stats/inlinechart/index.jsx

```diff
@@ -0,0 +1,93 @@
+// @flow
+
+import React, {Component} from 'react';
+import c3 from 'c3';
+import 'whatwg-fetch';
+import 'core-js/es6/promise';
+import moment from 'moment';
+import type {Props, State, RefObject, Data} from './types/index';
+
+class InlineChart extends Component<Props, State> {
+  c3ChartContainer: RefObject
+
+  constructor(props: Props) {
+    super(props);
+    this.c3ChartContainer = React.createRef();
+    this.state = {
+      loading: true,
+      title: this.props.title,
+      total: '',
+      values: [],
+      unit: '',
+      unitPluralized: this.props.unitPluralized
+    };
+  }
+
+  generateC3Chart(){
+    c3.generate({
+      bindto: this.c3ChartContainer.current,
+      axis: {
+        x: {show:false},
+        y: {show:false}
+      },
+      legend: { show: false },
+      point: { show: false },
+      data: {
+        columns: [
+          [...this.state.values]
+        ]
+      },
+      size: { height: 60 },
+      tooltip: {
+        contents: function (d) {
+          return `<span><i class='tooltip-dot'></i> ${d[0].value}</span>`;
+        }
+      }
+    });
+  }
+
+  updateState(data: Data){
+    let unit = data.metric.unit;
+    if (unit.slice(-1) !== 's' && data.total !== 1) {
+      unit = this.state.unitPluralized;
+    }
+    const total = `${data.total.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",")} ${unit.substring(0, 10)}`;
```

Comment:
Mighgt be worth it to separate the string crafting happening here in a different fn with some relevant naming